pkg/utils/requiredfieldsvalidator/validator.go

```diff
@@ -0,0 +1,147 @@
+package requiredfieldsvalidator
+
+import (
+	"fmt"
+	"reflect"
+	"strings"
+)
+
+const (
+	omitemptyTag = "omitempty"
```

Comment:
I suggest you to add comma to the value `omitempty` because the current implementation skips even if there is no omitempty constraint, but there is an omitempty field name. 
For instance we have the following struct:
```go
type t struct {
  Omitempty string `json:"omitempty"` // this field considered to be required because there is no an omitempty 
                                      // constraint, but the current implementation doesn't validates the following one because 
                                      // there is a json field name omitempty
}
```

Also I think this is not a tag, but a constraint `omitemptyConstraint`

Reply:
agree, fixed